play/src/pusher/controllers/AuthenticateController.ts

```diff
@@ -420,14 +419,16 @@ export class AuthenticateController extends BaseHttpController {
 
             const authToken = jwtTokenManager.createAuthToken(email || userUuid);
 
-            res.json({
-                authToken,
-                userUuid,
-                email,
-                roomUrl,
-                mapUrlStart,
-                organizationMemberToken,
-            } as RegisterData);
+            res.json(
+                isRegisterData.parse({
+                    authToken,
+                    userUuid,
+                    email,
+                    roomUrl,
+                    mapUrlStart,
+                    organizationMemberToken,
+                })
```

Comment:
Nope, you need to keep the "as" here.
Using Zod is wrong for this.
We want to ensure at COMPILE TIME the code is wrong. Not at runtime.